skglm/tests/test_prox_newton.py

```diff
@@ -37,14 +37,17 @@ def test_pn_vs_sklearn(rho, X_density):
     alpha = rho * alpha_max
 
     sk_log_reg = LogisticRegression(penalty='l1', C=1/(n_samples * alpha),
-                                    fit_intercept=False, tol=1e-9, solver='liblinear')
+                                    fit_intercept=fit_intercept,
```

Comment:
liblinear penalizes intercept IIRC. Does it work fine here?

Reply:
No it doesn't work here. The test against ``sklearn`` fails.
I was actually scratching my head to figure out why. Thanks @mathurinm for reminding me that liblinear fits an unpenalized intercept.

Any ideas against whom should we uni test?

Reply:
as a first try i'd test `intercept_scaling=1000`  in sklearn logreg it it doesn't break convergence (lower value else)

Reply:
> liblinear fits an unpenalized intercept

Did you mean 'liblinear fits a penalized intercept' ? https://github.com/scikit-learn/scikit-learn/issues/10626

Reply:
> Any ideas against whom should we uni test?

Could we compare to our own cd solvers with intercept ?

Reply:
yes penalized, my bad
just try a high intercept scaling first to see if you get a value within reasonable tolerance of our unpenalized intercept

Reply:
I checked against [``scikit-learn:saga``](https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression:~:text=The%20following%20table%20summarizes%20the%20penalties%20supported%20by%20each%20solver%3A) as it fits unpenalized intercept